application/forms/IcingaDependencyForm.php

```diff
@@ -212,11 +212,18 @@ protected function addObjectsElement()
 
         if (!empty($sentParent) || $dependency->isApplyRule()) {
             $parentService = $dependency->get('parent_service');
+            if ($parentService === null) {
+                $parentServiceVar = $dependency->get('parent_service_by_name') ?? '';
+                if (strlen($parentServiceVar) > 0) {
+                    $parentService = '$' . $dependency->get('parent_service_by_name') . '$';
+                }
+            }
             $this->addElement('text', 'parent_service', [
                     'label' => $this->translate('Parent Service'),
                     'description' => $this->translate(
                         'Optional. The parent service. If omitted this dependency'
-                         . ' object is treated as host dependency.'
+                        . ' object is treated as host dependency. You might want to refer'
```

Comment:
*refer to*

Reply:
This shouldn't be suggested in case of host dependencies, as for them this is not working. So, only suggest this for service dependencies and add an validator in any case which fails if the user still tries to do this for host dependencies.